app/public/js/localSprites.js

```diff
@@ -141,6 +144,35 @@ function updateLocalSprite(localSprite) {
         // This flag will be used in the update loop to remove this sprite from the list of localSprites.
         localSprite.shouldBeRemoved = true;
     }
+    //geomtry collision checks for projectiles that die on impact
+    //BROKEN: It looks to me like "alignToTile" isn't happening, and the fireball is detonating in the frame before it would impact something,
+    //  rather than right next to the wall it would be moving past.
+    if (localSprite.vx && localSprite.diesOnImpact) {
+        if (localSprite.vx < 0) {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vx, TILE_LEFT))) localSprite.shouldBeRemoved = true;
```

Comment:
You don't need these outer parentheses just do:
`if (!moveSpriteInDirection(localSprite, localSprite.vx, TILE_LEFT))`
here and below.